content/docs/rfcs/43/README.md

```diff
@@ -0,0 +1,347 @@
+where `n` is serialized as `uint64`.
+
+In this way, sender's and recipient's devices 
+can keep updated a buffer of `messageNametag` to sieve 
+while listening to messages sent over `/{application-name}/{application-version}/wakunoise/1/sessions-{shard-id}/` (i.e., the next 50 not yet seen). 
+They will then be able to further identify if one or more messages were eventually lost 
+or not-yet-delivered during the communication.
+This approach brings also the advantage that 
+communicating devices can efficiently identify encrypted messages addressed to them.
+
+We note that since the `ChaChaPoly` cipher used to encrypt messages supports *additional data*, 
+an encrypted payload can be further authenticated by passing the `messageNametag` as additional data to the encryption/decryption routine. 
+In this way, an attacker would be unable to craft an authenticated Waku message 
+even in case the currently used symmetric encryption key is compromised, 
+unless `ctsInbound`, `ctsOutbound` or the `messageNametag` buffer lists were compromised too.
+
+# Security/Privacy Considerations
+
+### Assumptions
+
+- The attacker is active, i.e. can interact with both devices `A` and `B` by sending messages over `contentTopic`.
+
+- The attacker has access to the QR code, that is knows the ephemeral key `eB`, the commitment `H(sB||r)` and the `contentTopic` exposed by the device `B`.
+
+- Devices `A` and `B` are considered trusted (otherwise the attacker will simply exfiltrate the relevant information from the attacked device).
+
+- As common for Noise, we assume that ephemeral keys cannot be compromised, while static keys might be later compromised. However, we enforce in the pairing phase extra security mechanisms (i.e. use of commitments for static keys) that will prevent some attacks possible when ephemeral keys are weak or get compromised.
+ 
+### Rationale
+
+- The device `B` exposes a commitment to its static key `sB` because:
+    - if the private key of `eB` is weak or gets compromised, an attacker can impersonate `B` by sending in message `c.` to device `A` his own static key and successfully complete the pairing phase. Note that being able to compromise `eB` is not contemplated by our security assumptions.
+    - `B` cannot adaptively choose a static key based on the state of the Noise handshake at the end of message `b.`, i.e. after the authentication code is confirmed. Note that device `B` is trusted in our security assumptions.
+    - Confirming the authentication code after processing message `b.` will ensure that no Man-in-the-Middle (MitM) can send a static key different than `sB`.
+
+- The device `A` sends a commitment to its static key `sA` because:
+    - `A` cannot adaptively choose a static key based on the state of the Noise handshake at the end of message `b.`, i.e. after the authentication code is confirmed. Note that device `A` is trusted in our security assumptions.
```

Comment:
If `A` is trusted, then why should it want to change its static key at a later stage?

Reply:
Quoting your reply from the prior PR
> That's the purpose of the note, that is we're giving extra protection. Without commitments (and authcode) the first exchanges are completely not authenticated, hence B could send arbitrary static key to, for example:
> get A's static key without revealing its real one (note that the hash can be binded in ZK to seed phrase to show the static key to come from same source of trust);
> have an encrypted payload where the MSB bytes are set to a known value with consequences on anonymity;
etc.

Thanks for your explanation. 
To me, B's static key is an arbitrary choice until before the pairing session starts. B has every right to pick any static key as long as it uses the same during the entire interaction. Unless there are some other objectives that are not mentioned in the text.
I think the commitment idea is tightly bound to what `adaptive static key selection` means. The commitment is necessary to prevent the other end to manipulate its static keys for `some reason`. But that `reason` is not explained. Before starting the protocol, A has no clue about Bob's static key, then why A should care about Bob picking any static key even adaptively? Are we seeking something more than making a private and authenticated communication channel between two mutually trusted parties?

Reply:
I see your point, just the assumptions and the adversarial power is somewhat unclear.  

In my view, I see the following adversarial powers:
1) An attacker who eavesdrops on the QR code
2) An attacker who eavesdrops and injects data during the QR code 
3) a passive MIM  attacker (in a p2p network) 
4) a active MIM  attacker (in a p2p network) 
5) An attacker that compromises eA and eB
6) An attacker can compromise sA and sB


The commitment of B on sB||r is to provide anonymity against adversary # 1
The commitment on H(sA||s) is to protect A's anonymity against an attacker with the capability of 2,4, and 5

And the current protocol design only considers an adversary who is capable of # 1, # 3 and # 4.